### PR TITLE
2/n - Make Gemma use regular TransformerDecoder

### PR DESCRIPTION
**Pull request by felipemello1**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (please add here)

This PR was built on top of https://github.com/pytorch/torchtune/pull/1547#issuecomment-2345166333, so qwen changes should disappear from here after the other PR is merged

#### Changelog
- updates gemma to use TransformerDecoder
- Uses TiedLinear for the output projection
- Uses Normalized Embedding to put together embedding + gemma norm
- Also quick fix on lora alpha to be 2x lora rank
- Reduce warmup steps to 10. I dont see why we need 100. LoRA training should be stable, we dont have it in finetuning. I did this after comparing a short run of full vs lora, and realizing that lora loss didnt decrease at all for the first 50 steps.

#### Test plan

resume from checkpoint working well

```
tune run --nnodes 1 --nproc_per_node 8 full_finetune_distributed --config gemma/2B_full batch_size=8 max_steps_per_epoch=20 metric_logger=torchtune.training.metric_logging.WandBLogger gradient_accumulation_steps=1 epochs=2 compile=True
```
image

### CODE REVIEW

**Review by joecummings — COMMENTED** [review]
🫡

File: torchtune/training/_compile.py [hunk 1]
```diff
@@ -5,32 +5,27 @@
...
+from torchtune.modules import TransformerDecoder, TransformerSelfAttentionLayer
 from torchtune.modules.loss import CEWithChunkedOutputLoss
 from torchtune.utils import get_logger, torch_version_ge
 
 log = get_logger("INFO")
 
 
 def compile_model(
-    model: Union[TransformerDecoder, TiedEmbeddingTransformerDecoder],
+    model: TransformerDecoder,
```
[the remark this answers is not in the document]
Comment (felipemello1) [reply torchtune/training/_compile.py:20]: I forgot and left a comment that i was doing it in 2/n

File: torchtune/models/gemma/gemma_norm_embedding.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import torch.nn as nn
+
+
+class GemmaNormEmbeddings(nn.Embedding):
+    def __init__(self, in_dim: int, out_dim: int):
```
Comment (RdoubleA) [inline torchtune/models/gemma/gemma_norm_embedding.py:12]: docstrings, esp to explain why this is a separate class

**Review by RdoubleA — APPROVED** [review]
